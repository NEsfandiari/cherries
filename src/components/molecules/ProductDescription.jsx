import React, { Component } from 'react'
import styled from 'styled-components'
import {
  StyledHr,
  StyledButton,
  QuantityAdjustButton,
  ProductDetail,
} from '../atoms'
import { ProductOrder } from '../molecules'

const Layout = styled.div`
  flex-basis: 45%;
  display: flex;
  flex-direction: column;
  padding: 1rem;

  .purchase {
    display: flex;
    align-items: center;
    width: 100%;
    justify-content: flex-start;
    @media (max-width: 420px) {
      flex-direction: column;
    }
  }
  .price {
    font-size: 0.9rem;
    text-align: center;
    padding: 0.5rem;
    margin: 0;
    color: #ff009a;
    height: 2.5rem;
    width: 5rem;
  }
  h1 {
    white-space: nowrap;
    overflow: visible;
    width: 23rem;
  }

  @media (max-width: 420px) {
    padding: 0rem;
    hr {
      margin-left: 0;
      margin-right: 0;
    }
  }
`

class ProductDescription extends Component {
  state = {
    quantity: 1,
    price: this.props.price,
    status: 'ADD TO BAG',
    oneTimePurchase: true,
  }
  handleChange = e => {
    e.preventDefault()
    this.setState({ [e.target.name]: e.target.value })
  }
  handleAdjust = e => {
    let newQuantity
    if (e.target.className === 'add') {
      newQuantity = this.state.quantity + 1
      this.setState({ quantity: newQuantity })
      this.handlePrice(newQuantity)
    } else {
      if (this.state.quantity > 1) {
        newQuantity = this.state.quantity - 1
        this.setState({ quantity: newQuantity })
        this.handlePrice(newQuantity)
      }
    }
  }
  handlePrice = quantity => {
    const newPrice = quantity * this.props.price
    this.setState({
      price: newPrice.toFixed(2),
    })
  }
  handleSubmit = e => {
    e.preventDefault()
    this.props.handleCart(
      'add',
      this.props.title,
      this.props.price,
      parseInt(this.state.quantity),
      this.props.images[0],
      this.props.sku
    )
    console.log('item added')
    this.setState({ status: 'ADDED!' })
  }
  render() {
    const productCopy = this.props.productCopy.map((statement, i) => {
      if (statement.content.length === 1)
        return <p key={i}>{statement.content[0].value}</p>
      else {
        // Turn Special marks into subsequent html modifiers
        let fullClause = []
        statement.content.forEach(clause => {
          if (clause.marks.length > 0) {
            if (clause.marks[0].type === 'bold')
              fullClause.push(<b>{clause.value}</b>)
          } else fullClause.push(clause.value)
        })
        return <p key={i}>{fullClause}</p>
      }
    })
    return (
      <Layout>
        <h1>{this.props.title}</h1>
        {productCopy}
        <StyledHr width={'100%'} margin={'.8rem'} />
        <div>
          <ProductOrder
            price={this.state.price}
            handleAdjust={this.handleAdjust}
            handleChange={this.handleChange}
          />
        </div>
        <form className="purchase" onSubmit={this.handleSubmit}>
          <QuantityAdjustButton
            quantity={this.state.quantity}
            handleAdjust={this.handleAdjust}
            id={''}
            color={'#E20031'}
            height={'2.5rem'}
            width={'3rem'}
          />
          <StyledButton height={'2.5rem'} width={'18rem'} fontSize={'.65rem'}>
            <b>{this.state.status}</b>
          </StyledButton>
        </form>
        <ProductDetail details={this.props.details} />
      </Layout>
    )
  }
}

export default ProductDescription
