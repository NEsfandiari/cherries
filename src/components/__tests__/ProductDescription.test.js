import React from 'react'
import renderer from 'react-test-renderer'
import ProductDescription from '../molecules/ProductDescription'

describe('ProductDescription', () =>
  it('renders correctly', () => {
    const price = '9.99'
    const title = 'Test Product'
    const images = [
      {
        file: {
          url:
            '//images.ctfassets.net/4yejk6rho2rp/1seQeOcUKEqkqG8iMeyQkO/f314700fc9f0e8d57715c30853a7859c/Retro-cylindrical-geometry-wood-earrings-jewelry-exaggerated-Christmas-red-green-colours-wood-pendant-earrings-for-women.jpg',
        },
      },
      {
        file: {
          url:
            '//images.ctfassets.net/4yejk6rho2rp/7cx6GGghKo6ukOwq0ceoqi/150babae3d22e18cdb3fc87ec45564ac/Retro-cylindrical-geometry-wood-earrings-jewelry-exaggerated-Christmas-red-green-colours-wood-pendant-earrings-for-women.jpg',
        },
      },
      {
        file: {
          url:
            '//images.ctfassets.net/4yejk6rho2rp/4rSP9SxdLqaEaUI62GMsek/70ceaedde47a68c6152481aa12f91327/Retro-cylindrical-geometry-wood-earrings-jewelry-exaggerated-Christmas-red-green-colours-wood-pendant-earrings-for-women.jpg',
        },
      },
    ]
    const sku = '81'
    const productCopy = [
      {
        nodeType: 'paragraph',
        content: [
          {
            value: 'Hi',
            nodeType: 'text',
          },
        ],
      },
    ]

    const details = [
              'Enamel pin semiotics mixtape art party, selvage farm-to-table shabby chic vape 3 wolf moon sriracha glossier disrupt cronut celiac. Whatever vegan hexagon occupy celiac poutine enamel pin fanny pack tattooed master cleanse squid salvia. Celiac flexitarian biodiesel, coloring book raclette glossier migas. Fanny pack af tousled master cleanse woke. Jianbing whatever echo park, salvia raclette dreamcatcher narwhal letterpress pour-over.',
    ]

    const tree = renderer
      .create(
        <ProductDescription
          price={price}
          title={title}
          images={images}
          sku={sku}
          productCopy={productCopy}
          details={details}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  }))
