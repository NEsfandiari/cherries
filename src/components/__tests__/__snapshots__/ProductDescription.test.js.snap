// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDescription renders correctly 1`] = `
<div
  className="sc-iAyFgw ihxKmm"
>
  <h1>
    Test Product
  </h1>
  <p>
    Hi
  </p>
  <hr
    className="sc-dnqmqq dAdrog"
    width="100%"
  />
  <div>
    <div
      className="sc-jKJlTe bFiEdw"
    >
      <form
        action="#"
      >
        <p>
          <input
            className="purchaseSelector"
            id="oneTimePurchase"
            name="radio-group"
            type="radio"
          />
          <label
            for="oneTimePurchase"
          >
             
            <span
              className="purchase-text"
            >
               One-Time Purchase:
            </span>
            <strong
              onClick={[Function]}
            >
              9.99
            </strong>
          </label>
        </p>
        <p>
          <input
            id="subscription"
            name="radio-group"
            type="radio"
          />
          <label
            className="purchaseSelector"
            for="subscription"
          >
             
            <span
              className="purchase-text"
            >
              Subscribe and Save:
            </span>
             
            <strong>
              $10/mo
            </strong>
          </label>
        </p>
      </form>
    </div>
  </div>
  <form
    className="purchase"
    onSubmit={[Function]}
  >
    <div
      className="sc-iwsKbI laXdmC"
      color="#E20031"
      height="2.5rem"
    >
      <div
        className="sub"
        data-id=""
        onClick={[Function]}
      >
        -
      </div>
      <p>
        1
      </p>
      <div
        className="add"
        data-id=""
        onClick={[Function]}
      >
        +
      </div>
    </div>
    <button
      className="sc-gzVnrw dTowVe"
      fontSize=".65rem"
      height="2.5rem"
      width="18rem"
    >
      <b>
        ADD TO BAG
      </b>
    </button>
  </form>
  <div
    className="sc-gZMcBi acVHm"
  >
    <h4>
      Details
    </h4>
    <p>
      Enamel pin semiotics mixtape art party, selvage farm-to-table shabby chic vape 3 wolf moon sriracha glossier disrupt cronut celiac. Whatever vegan hexagon occupy celiac poutine enamel pin fanny pack tattooed master cleanse squid salvia. Celiac flexitarian biodiesel, coloring book raclette glossier migas. Fanny pack af tousled master cleanse woke. Jianbing whatever echo park, salvia raclette dreamcatcher narwhal letterpress pour-over.
    </p>
  </div>
</div>
`;
